### DISCUSSION TITLE
Introducing entities concept

### DISCUSSION THREAD

**Discussion by serrq** [mullvad/mullvadvpn-app#9125] [Feature requests & Ideas]
### I have checked if others have suggested this already

- [X] I have checked this issue tracker to see if others have reported similar issues.

### Feature description

Currently we have one account that allows up to five devices. So basically one device is equal to one license (of the five on total).

My idea introduce a different approach to have multiple licence in a same device.

Why?

Firstly because I have one device and leave 4 licences unused.

Even important because with "entities" I can create up to five "profiles" (entities) in the same device and put apps in the right drawer.

For example I set 4 entities (licences) in my smartphone. Each entity is fully independent from the other ones. It allows starting compartmentalization already in network.

• entity one: bank 
• entity two: spotify, netflix, etc.
• entity three: telegram, whatsapp, signal, etc.
• entity four: spare use or what you want

All platforms involved.
Each entity has its IP, server, etc. Fully isolate one from other ones.

### Alternative solutions

I don't know.

### Type of feature

- [X] Better privacy/anonymity
- [ ] Better at circumventing censorship
- [ ] Easier to use
- [ ] Other

### Operating System

- [X] Android
- [X] iOS
- [X] Windows
- [X] macOS
- [X] Linux

**Comment by Serock3** [comment]
Hello and thanks for the suggestion 🙂 This would be a cool feature that I would like to see myself, and it's similar to other requests we have gotten in the past. It may be possible using namespaces on Linux, far more challenging on mobile platforms where we don't set the routing ourselves. It's an ambitious feature and we have many other ones with higher priority, so we likely won't work on it anytime soon. I brought it up with the team and we'll keep it in mind.

**Comment by gsture** [comment]
You could use a work profile or use separate user profiles. All those profiles can have a separate vpn connection. This also comes with the extra benefit of having the user data from those apps separated. You could also use a socks5 proxy over your vpn to get another identity for selected apps.

**Comment by MrChocolatine** [comment]
> You could use a setup work profile or use separate user profiles. All those profiles can have a separate vpn connection. This also comes with the extra benefit of having the user data from those apps separated. You could also use a socks5 proxy over your vpn to get another identity for selected apps.

+1

As an example, I use GrapheneOS and my 5 licenses (devices) are used via different User Profiles.
